lib/rides/pages/search_deals_page.dart

```diff
@@ -0,0 +1,291 @@
+    return '${date.day}.${date.month}.${date.year}';
+  }
+
+  String _formatTime(DateTime time) {
+    return DateFormat.Hm().format(time);
+  }
+
+  String? _timeValidator(String? value) {
+    if (value == null || value.isEmpty) {
+      return 'Please enter a time';
+    }
+    if (_selectedDate.isBefore(_firstDate)) {
+      return 'Please enter a valid time';
+    }
+    return null;
+  }
+
+  //todo: get possible Rides from Algorithm
+  void getRides() async {
+    List<Ride> initializes = await supabaseClient
+        .from('drives').select()
+        .eq('start', _startController.text)
+        .order('start_time', ascending: true)
+        .then((drive)
+    => Drive.fromJsonList(drive).map((e)
+    => e.toRide(_startController.text,
+      _destinationController.text,
+      e.startTime,
+      e.endTime,
+      _dropdownValue,
+      -1,
+      10.25,)).toList());
+    setState(() {
+      ridesuggestion = initializes;
+    });
+  }
+
+  //todo: filter
+  void filter(){
+    getRides();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Search Ride'),
+      ),
+      body: Padding(
+        padding: const EdgeInsets.all(8.0),
+        child: Column(
+          children: [
+            FixedTimeline(
+              theme: CustomTimelineTheme.of(context),
+              children: [
+                TimelineTile(
+                    contents: Padding(
+                      padding: const EdgeInsets.all(4.0),
+                      child: Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                        children: [
+                          Expanded(
+                            child: ElevatedButton(
+                              onPressed: () async {
+                                final AddressSuggestion? startSuggestion =
+                                  await showSearch<AddressSuggestion?>(
+                                  context: context,
+                                  delegate: AddressSearchDelegate(),
+                                  query: _startController.text,
+                                  );
+                                 if (startSuggestion != null) {
+                                    _startController.text = startSuggestion.name;
+                                    getRides();
+                                 }
+                              },
+                              child: Align(
+                                  alignment: Alignment.centerLeft,
+                                  child: Text(_startController.text),
+                              ),
+                            ),
+                          ),
+                          const SizedBox(width: 50),
+                          SizedBox(
+                            width: 45,
+                            child: TextFormField(
+                              readOnly: true,
```

Comment:
```suggestion
                              decoration: const InputDecoration(
                      border: OutlineInputBorder(),
                      labelText: "Date",
                    ),
                    readOnly: true,
```
Or Maybe something else but i think it would be better if these to Field would be labeled somehow.

Reply:
Find ich auch, aber dann verzieht sich das mit der Ansicht. Ich schau das ich was finde